pyramus/src/main/java/fi/otavanopisto/pyramus/views/applications/ApplicationUtils.java

```diff
@@ -126,6 +167,85 @@ public static String genderUiValue(String value) {
+    if (!StringUtils.isBlank(ssnEnd) && ssnEnd.length() == 4) {
+      char c = ssnEnd.charAt(2);
+      return c == '1' || c == '3' || c == '5' || c == '7' || c == '9' ? Sex.MALE : Sex.FEMALE;
+    }
+    return "nainen".equals(genderValue) ? Sex.FEMALE : Sex.MALE;
+  }
+  
+  public static StudentExaminationType resolveStudentExaminationType(String examinationType) {
+    StudentExaminationTypeDAO studentExaminationTypeDAO = DAOFactory.getInstance().getStudentExaminationTypeDAO();
+    switch (examinationType) {
+    case "muu":
+      return studentExaminationTypeDAO.findById(1L); // Muu tutkinto
+    case "ammatillinen-perus":
+      return studentExaminationTypeDAO.findById(2L); // Ammatillinen perustutkinto
+    case "ammatillinen-korkea":
+      return studentExaminationTypeDAO.findById(3L); // Ammattikorkeakoulututkinto
+    case "kaksoistutkinto":
+      return studentExaminationTypeDAO.findById(4L); // Kaksoistutkinto
+    case "yo-tutkinto":
+      return studentExaminationTypeDAO.findById(5L); // YO-tutkinto / lukion oppimäärä
+    case "oppisopimus":
+      return studentExaminationTypeDAO.findById(6L); // Oppisopimuskoulutus
+    default:
+      return null;
+    }
+  }
+  
+  public static StudentActivityType resolveStudentActivityType(String activityType) {
+    StudentActivityTypeDAO studentActivityTypeDAO = DAOFactory.getInstance().getStudentActivityTypeDAO();
+    switch (activityType) {
+    case "tyollinen":
+      return studentActivityTypeDAO.findById(1L); // Työllinen
+    case "tyoton":
+      return studentActivityTypeDAO.findById(2L); // Työtön
+    case "opiskelija":
+      return studentActivityTypeDAO.findById(3L); // Opiskelija
+    case "elakelainen":
+      return studentActivityTypeDAO.findById(4L); // Eläkeläinen
+    case "muu":
+      return studentActivityTypeDAO.findById(5L); // Muu
+    default:
+      return null;
+    }
+  }
+  
+  public static StudyProgramme resolveStudyProgramme(String line, String foreignLine) {
+    StudyProgrammeDAO studyProgrammeDAO = DAOFactory.getInstance().getStudyProgrammeDAO();
+    switch (line) {
+    case "aineopiskelu":
+      return studyProgrammeDAO.findById(13L); // Internetix/lukio
+    case "nettilukio":
+      return studyProgrammeDAO.findById(6L); // Nettilukio
+    case "nettipk":
+      return studyProgrammeDAO.findById(7L); // Nettiperuskoulu
+    case "aikuislukio":
+      return studyProgrammeDAO.findById(1L); // Aikuislukio
+    case "bandilinja":
+      return studyProgrammeDAO.findById(8L); // Bändilinja/vapaa
+    case "kasvatustieteet":
+      return null;
+    case "laakislinja":
+      return studyProgrammeDAO.findById(31L); // Lääketieteen opintoihin valmentava koulutus
+    case "mk":
+      switch (foreignLine) {
+      case "apa":
+        return studyProgrammeDAO.findById(29L); // Aikuisten perusopetuksen alkuvaiheen opetus
+      case "pk":
+        return studyProgrammeDAO.findById(15L); // Monikulttuurinen peruskoululinja
```

Comment:
Double check 15 - it's 11 for some...

Reply:
Yup, have to cross-reference against old application forms (ping @petja-laitila)